src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let currentState = state;
```

Comment:
Instead of directly assigning the `state` to `currentState`, you should create a copy of the `state` to avoid mutating the original `state` object. Use the spread operator or `Object.assign` to create a shallow copy.